geniza/corpus/solr_queryset.py

```diff
@@ -219,6 +226,19 @@ def keyword_search(self, search_term):
             )
         return search
 
+    def regex_search(self, search_term):
+        """Build a Lucene query for searching with regex."""
+        # store original regex query
+        original_regex = search_term
```

Comment:
Do you mean things like reserved characters? I don't know if there are ways to actually break it in terms of producing server errors—solr just seems to produce 0 results when the query is malformed. But definitely open to implementing form validation if you have recommendations for the kinds of things I should be validating against.

Reply:
I was thinking primarily of malformed regex characters. I don't think there's a security risk in this case but seems like the kind of input where there's potential for injection.  You could maybe try compiling it as a python regex but I'm not sure if that's warranted.

Ideally, I think you want the user to be able to easily understand the difference between zero results because nothing was found and zero results because the query wasn't valid.

Reply:
This ended up being a big can of worms! I found what I believe to be the most likely ways to break it, but I'm sure I missed a lot. I'm not sure what the best next steps are for this, but the form validation does quite a lot now. Thanks for the suggestion because I would have totally missed this otherwise!

Reply:
I am also wondering if this is fully possible. We can check some cases, but there are bound to be many that slip through the cracks, and validating the regex is probably ultimately up to Java/Lucene.

Is there a way to catch the specific solr error when it's a 500 from parasolr? The only place I was able to catch an exception was here:
https://github.com/Princeton-CDH/geniza/blob/d8f3e997e6997a10374335683136e3ab5ec15ce9/geniza/corpus/views.py#L352-L356

It didn't seem to raise any exception on making the actual search query, but the 500 error is visible in logs. If we can catch the specific error I can give more detailed regex validation feedback to the user after the fact.

Reply:
I don't think you can catch everything but it sounds like you've done a good effort to catch a number of likely things.

If there's a change that would be useful on the parasolr side, we could consider that. Does it make sense to create an issue on the parasolr repo with an example and the desired/imagined behavior?

Reply:
Sure, I can do that!

Reply:
Done, let me know if any more details would be helpful: https://github.com/Princeton-CDH/parasolr/issues/88